staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1005,6 +1164,40 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		waitGroup:           &utilwaitgroup.SafeWaitGroup{},
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		var processedCh, errCh chan struct{}
+		ctx := apirequest.NewContext()
+		var retry int
+		retries := 3
+
+		done := false
+		for retry < retries {
+			// spawn storage version update in background and use channels to make handlers wait
+			processedCh = make(chan struct{})
+			errCh = make(chan struct{})
+			err = r.storageVersionManager.UpdateStorageVersion(ctx, crd, nil, processedCh, errCh)
+			select {
+			case <-errCh:
+				klog.Errorf("retry %d, failed to update storage version for %v : %v", retry, crd, err)
+				retry++
+				continue
+			case <-processedCh:
+				done = true
+			}
+
+			if done {
+				break
+			}
+		}
+
+		ret.storageVersionUpdate = &storageVersionUpdate{
+			processedCh: processedCh,
+			errCh:       errCh,
+			timeout:     time.Now().Add(storageVersionUpdateTimeout),
```

Comment:
Does this field get reset for each CRD? Do the channels represent just updating for the single SV, or for all SVs?